### PR TITLE
Doxygen documentation2

### PR DESCRIPTION
**Pull request by endJunction**
continuation of #1569 .

### CODE REVIEW

File: BaseLib/StringTools.h [hunk 1]
```diff
@@ -22,10 +22,10 @@
 #include <vector>
 
 namespace BaseLib {
-
 /**
- *  Splits a string into a vector of strings. This method only works for string seperation
- *  recognised by the std::stringstream iterator such as ' ' or '\t'.
+ *  Splits a string into a vector of strings. This method only works for string
+ *  seperation recognised by the std::stringstream iterator such as ' ' or
```
Comment (chleh) [inline BaseLib/StringTools.h]: typo separation

File: GeoLib/OctTree.h [hunk 2]
```diff
@@ -86,6 +86,8 @@ class OctTree {
     /// private constructor
     /// @param ll lower left point
     /// @param ur upper right point
+    /// @param eps the euclidean distance as a threshold to make objects unique
+    /// [default std::numeric_limits<double>::epsilon()]
```
Comment (chleh) [inline GeoLib/OctTree.h]: Where is the default set? The method declaration does not declare a default value.

File: GeoLib/QuadTree.h [hunk 3]
```diff
@@ -46,6 +46,7 @@ template <typename POINT> class QuadTree
      * (lower left and the upper right points).
      * @param ll lower left point of the square
      * @param ur upper right point of the square
+     * @param max_points_per_node maximum number of points per leaf
```
Comment (chleh) [inline GeoLib/QuadTree.h]: node vs. leaf?

File: MeshLib/Elements/PyramidRule5.h [hunk 4]
```diff
@@ -80,12 +80,11 @@ class PyramidRule5 : public CellRule
     static const Element* getFace(const Element* e, unsigned i);
 
     /**
-     * Checks if a point is inside the element.
-     * @param pnt a 3D MathLib::Point3d object
-     * @param eps tolerance for numerical algorithm used or computing the property
-     * @return true if the point is not outside the element, false otherwise
+     * \copydoc MeshLib::Element::isPntInElement()
+     * @param nodes the nodes of the element.
```
Comment (chleh) [inline MeshLib/Elements/PyramidRule5.h]: You only document one parameter. Are the other parameter docs inherited by copydoc?
Smiliar for the other `isPntInElement()` methods.

**Comment by chleh** [comment]
Just recognized that some of my comments acutally apply to #1569.

File: GeoLib/QuadTree.h (hunk 3 above)
Comment (endJunction) [inline GeoLib/QuadTree.h]: Fixing the variable name.

File: MeshLib/Elements/PyramidRule5.h (hunk 4 above)
Comment (endJunction) [inline MeshLib/Elements/PyramidRule5.h]: Yes. Other parameters docs are copied (probably must have exactly the same types and names, which they do). Only extra parameters needs documentation.

**Pull request merged**